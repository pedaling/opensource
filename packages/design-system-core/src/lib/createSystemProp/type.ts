import type { Theme } from '../types';

export type InterpolationFn = (props: Record<string, any>) => Record<string, any>;

export type SystemPropConfig = {
  property: string;
  styleProperty?: string;
  scale?: keyof Pick<Theme, 'colors' | 'space' | 'opacity'>;
  disabled?: boolean;
  transform?: (value: any) => Record<string, any> | undefined;
};

export type SystemProp = {
  (props: Record<string, any>): Record<string, any>;
  propName: string;
  disabled: boolean;
};
