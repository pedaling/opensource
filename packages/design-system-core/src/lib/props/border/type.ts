import type { ResponsiveValue } from '../../types';

export type BorderProps = {
  borderWidth?: ResponsiveValue<number>;
  borderStyle?: ResponsiveValue<'solid' | 'none'>;
  borderColor?: ResponsiveValue<string>;
  borderTopWidth?: ResponsiveValue<number>;
  borderTopStyle?: ResponsiveValue<'solid' | 'none'>;
  borderTopColor?: ResponsiveValue<string>;
  borderRightWidth?: ResponsiveValue<number>;
  borderRightStyle?: ResponsiveValue<'solid' | 'none'>;
  borderRightColor?: ResponsiveValue<string>;
  borderBottomWidth?: ResponsiveValue<number>;
  borderBottomStyle?: ResponsiveValue<'solid' | 'none'>;
  borderBottomColor?: ResponsiveValue<string>;
  borderLeftWidth?: ResponsiveValue<number>;
  borderLeftStyle?: ResponsiveValue<'solid' | 'none'>;
  borderLeftColor?: ResponsiveValue<string>;
  borderRadius?: ResponsiveValue<number | string>;
  borderTopLeftRadius?: ResponsiveValue<number | string>;
  borderTopRightRadius?: ResponsiveValue<number | string>;
  borderBottomLeftRadius?: ResponsiveValue<number | string>;
  borderBottomRightRadius?: ResponsiveValue<number | string>;
  outlineWidth?: ResponsiveValue<number>;
  outlineStyle?: ResponsiveValue<'solid' | 'none'>;
  outlineColor?: ResponsiveValue<string>;
  outlineOffset?: ResponsiveValue<number>;
};
