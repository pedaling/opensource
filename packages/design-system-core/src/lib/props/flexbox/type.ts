import type { ResponsiveValue } from '../../types';

export type FlexboxProps = {
  flex?: ResponsiveValue<string | number>;
  flexGrow?: ResponsiveValue<string | number>;
  flexShrink?: ResponsiveValue<string | number>;
  flexBasis?: ResponsiveValue<string | number>;
  flexDirection?: ResponsiveValue<'column' | 'column-reverse' | 'row' | 'row-reverse'>;
  flexWrap?: ResponsiveValue<'nowrap' | 'wrap' | 'wrap-reverse'>;
  alignContent?: ResponsiveValue<
    | 'space-around'
    | 'space-between'
    | 'space-evenly'
    | 'stretch'
    | 'center'
    | 'end'
    | 'flex-end'
    | 'flex-start'
    | 'start'
    | 'baseline'
    | 'normal'
  >;
  alignItems?: ResponsiveValue<
    | 'center'
    | 'end'
    | 'flex-end'
    | 'flex-start'
    | 'self-end'
    | 'self-start'
    | 'start'
    | 'baseline'
    | 'normal'
    | 'stretch'
  >;
  alignSelf?: ResponsiveValue<
    | 'center'
    | 'end'
    | 'flex-end'
    | 'flex-start'
    | 'self-end'
    | 'self-start'
    | 'start'
    | 'auto'
    | 'baseline'
    | 'normal'
    | 'stretch'
  >;
  justifyContent?: ResponsiveValue<
    | 'space-around'
    | 'space-between'
    | 'space-evenly'
    | 'stretch'
    | 'center'
    | 'end'
    | 'flex-end'
    | 'flex-start'
    | 'start'
    | 'left'
    | 'normal'
    | 'right'
  >;
  justifyItems?: ResponsiveValue<
    | 'center'
    | 'end'
    | 'flex-end'
    | 'flex-start'
    | 'self-end'
    | 'self-start'
    | 'start'
    | 'baseline'
    | 'left'
    | 'legacy'
    | 'normal'
    | 'right'
    | 'stretch'
  >;
  justifySelf?: ResponsiveValue<
    | 'center'
    | 'end'
    | 'flex-end'
    | 'flex-start'
    | 'self-end'
    | 'self-start'
    | 'start'
    | 'auto'
    | 'baseline'
    | 'left'
    | 'normal'
    | 'right'
    | 'stretch'
  >;
  order?: ResponsiveValue<number>;
};
