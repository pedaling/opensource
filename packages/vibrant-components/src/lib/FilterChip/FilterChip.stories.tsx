import type { ComponentMeta, ComponentStory } from '@storybook/react';
import { Icon } from '@vibrant-ui/icons';
import { FilterChip } from './FilterChip';

export default {
  title: 'FilterChip',
  component: FilterChip,
  args: {
    size: 'md',
    startIcon: <Icon.Clip.Regular />,
    endIcon: <Icon.ArrowTriangleDown.Regular />,
    children: 'FilterChip',
  },
} as ComponentMeta<typeof FilterChip>;

export const Basic: ComponentStory<typeof FilterChip> = props => <FilterChip {...props} />;

export const withHref: ComponentStory<typeof FilterChip> = props => (
  <FilterChip {...props} href="https://www.vibrant-design.com" />
);

export const withLongText: ComponentStory<typeof FilterChip> = props => (
  <FilterChip {...props} children={'Long Text '.repeat(100)} />
);
