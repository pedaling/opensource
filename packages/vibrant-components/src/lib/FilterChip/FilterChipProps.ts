import type { ReactElement, Ref } from 'react';
import type { ResponsiveValue, TextChildren } from '@vibrant-ui/core';
import { propVariant, withVariation } from '@vibrant-ui/core';
import type { PressableProps } from '../Pressable';

export type FilterChipProps = {
  ref?: Ref<any>;
  size: ResponsiveValue<'md' | 'sm'>;
  startIcon?: ReactElement;
  endIcon?: ReactElement;
  selected?: boolean;
  disabled?: boolean;
  onClick?: PressableProps['onClick'];
  children: TextChildren;
  testId?: string;
  href?: string;
  lineLimit?: ResponsiveValue<number>;
  maxWidth?: ResponsiveValue<number | string>;
};

export const withFilterChipVariation = withVariation<FilterChipProps>('FilterChip')(
  propVariant({
    props: [
      {
        name: 'size',
        responsive: true,
      },
      {
        name: 'startIcon',
        keep: true,
      },
      { name: 'endIcon', keep: true },
    ],
    variants: ({ size }) => {
      if (size === 'sm') {
        return {
          bodyLevel: 4,
          iconSize: 14,
          minHeight: 30,
          px: 8,
          py: 7,
          spacing: 3,
          rounded: 'lg',
        } as const;
      }

      return {
        bodyLevel: 2,
        iconSize: 16,
        minHeight: 38,
        p: 10,
        spacing: 4,
        rounded: 'xl',
      } as const;
    },
  }),
  propVariant({
    props: [
      {
        name: 'selected',
        default: false,
      },
      {
        name: 'disabled',
        default: false,
        keep: true,
      },
    ],
    variants: ({
      selected,
      disabled,
    }: {
      selected: FilterChipProps['selected'];
      disabled: FilterChipProps['disabled'];
    }) => {
      if (disabled) {
        return {
          color: 'onView3',
          backgroundColor: selected ? 'disable' : 'surface1',
        } as const;
      }

      if (selected) {
        return {
          color: 'onInverseSurface',
          backgroundColor: 'inverseSurface',
          overlayColor: 'onView1',
          interactions: ['hover', 'focus', 'active'],
        } as const;
      }

      return {
        color: 'onView1',
        backgroundColor: 'surface1',
        overlayColor: 'onView1',
        interactions: ['hover', 'focus', 'active'],
      } as const;
    },
  })
);
