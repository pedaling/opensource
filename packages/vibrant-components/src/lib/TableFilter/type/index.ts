export type StringFilterOperator = 'contains' | 'empty' | 'equals' | 'notContains' | 'notEmpty' | 'notEquals';

export type StringFilter = {
  value: string;
  operator: StringFilterOperator[];
};

export type Filter = StringFilter & { dataKey: string };
