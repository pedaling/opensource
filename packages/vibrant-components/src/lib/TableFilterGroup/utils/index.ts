import type { DateFilterOperator, MultiSelectFilterOperator, Option, StringFilterOperator } from '../types';

export const isValueRequiredOperator = (op: string) => op === 'empty' || op === 'notEmpty';

export const isStringFilterValid = (filter: { value: string; operator: StringFilterOperator }) =>
  Boolean(filter.value) || isValueRequiredOperator(filter.operator);

export const isDateFilterValid = (filter: { value: Date[]; operator: DateFilterOperator }) => {
  if (isValueRequiredOperator(filter.operator)) return true;

  if (filter.operator === 'between') return filter.value.length >= 2;

  return filter.value.length >= 1;
};

export const isMultiSelectFilterValid = (filter: { value: Option['value'][]; operator: MultiSelectFilterOperator }) =>
  Boolean(filter.value.length > 0) || isValueRequiredOperator(filter.operator);
