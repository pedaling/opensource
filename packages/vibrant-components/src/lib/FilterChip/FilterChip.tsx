import { cloneElement } from 'react';
import { Box } from '@vibrant-ui/core';
import { Body } from '../Body';
import { Pressable } from '../Pressable';
import { withFilterChipVariation } from './FilterChipProps';

export const FilterChip = withFilterChipVariation(
  ({
    startIcon,
    endIcon,
    children,
    bodyLevel,
    color,
    spacing,
    iconSize,
    testId,
    lineLimit,
    maxWidth = '100%',
    ...props
  }) => (
    <Pressable
      testId={testId}
      flexGrow={0}
      flexShrink={0}
      rounded="full"
      maxWidth={maxWidth}
      flexDirection="row"
      alignItems="center"
      {...props}
    >
      {startIcon && (
        <Box as="span" mr={spacing}>
          {cloneElement(startIcon, { size: iconSize, fill: color })}
        </Box>
      )}
      <Body level={bodyLevel} color={color} lineLimit={lineLimit} wordBreak={lineLimit ? 'break-all' : 'normal'}>
        {children}
      </Body>
      {endIcon && (
        <Box as="span" ml={spacing}>
          {cloneElement(endIcon, { size: iconSize, fill: color })}
        </Box>
      )}
    </Pressable>
  )
);
