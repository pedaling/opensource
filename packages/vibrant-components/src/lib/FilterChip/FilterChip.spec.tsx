import type { ReactRenderer } from '@vibrant-ui/utils/testing-web';
import { createReactRenderer } from '@vibrant-ui/utils/testing-web';
import { FilterChip } from './FilterChip';

describe('<FilterChip />', () => {
  const { render } = createReactRenderer();
  let renderer: ReactRenderer;

  describe('when href provided', () => {
    beforeEach(() => {
      renderer = render(
        <FilterChip size="md" href="https://www.vibrant-design.com">
          Click me
        </FilterChip>
      );
    });

    it('should render link with href attribute', () => {
      expect(renderer.getByRole('link').getAttribute('href')).toBe('https://www.vibrant-design.com');
    });
  });
});
