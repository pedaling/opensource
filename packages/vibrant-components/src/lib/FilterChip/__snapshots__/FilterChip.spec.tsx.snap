// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterChip /> when href provided match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  box-sizing: border-box;
  position: relative;
  overflow: hidden;
  cursor: pointer;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  z-index: 0;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 100%;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: flex-start;
  align-self: flex-start;
  min-height: 38px;
  padding: 10px;
  border-radius: 16px;
  background-color: #00000008;
}

.emotion-1 {
  text-align: left;
  color: #0c0c0c;
  -webkit-text-fill-color: #0c0c0c;
  text-fill-color: #0c0c0c;
  word-break: normal;
  font-size: 0.875rem;
  line-height: 1.125rem;
  font-weight: 400;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: absolute;
  left: 0;
  right: 0;
  top: 0;
  bottom: 0;
  background-color: #0c0c0c;
  pointer-events: none;
}

<div>
  <a
    class="emotion-0"
    data-testid="pressable"
    href="https://www.vibrant-design.com"
    style="opacity: 1; transition-property: opacity; transition-duration: 200ms; transition-timing-function: cubic-bezier(0.5, 1, 0.89, 1);"
  >
    <span
      class="emotion-1"
      data-testid="body"
    >
      Click me
    </span>
    <span
      class="emotion-2"
      style="opacity: 0; transition-property: opacity; transition-duration: 200ms; transition-timing-function: cubic-bezier(0.5, 1, 0.89, 1);"
    />
  </a>
</div>
`;
